### DISCUSSION TITLE
Software update asking if DFU is a zip?

### DISCUSSION THREAD

**Discussion by devinhenkel** [InfiniTimeOrg/InfiniLink#80] [Q&A]
Hey, all. Just got my PineTime today and super stoked to dig in. Installed InfiniLink from the App Store and connected no problem. However, it notified me that a firmware update (1.11) was available (mine shipped with 1.0.0), but it just sat on the update screen. I turned on debugging and the DFU log says 

![image](https://user-images.githubusercontent.com/209212/196259053-0e74b8af-e1b7-4d2f-b177-ae43377f726b.jpeg)

Any ideas?

Thanks. 

D.

**Comment by devinhenkel** [comment]
> Hey, all. Just got my PineTime today and super stoked to dig in. Installed InfiniLink from the App Store and connected no problem. However, it notified me that a firmware update (1.11) was available (mine shipped with 1.0.0), but it just sat on the update screen. I turned on debugging and the DFU log says 
> 
> ![image](https://user-images.githubusercontent.com/209212/196259053-0e74b8af-e1b7-4d2f-b177-ae43377f726b.jpeg)
> 
> Any ideas?
> 
> Thanks. 
> 
> D.

I actually see an issue switching the got branch from "master" to "main." Related? Is there a way to have a duplicate of the main branch called "master" so update will work until the changes to the app get pushed out?

**Reply by aafishman** [reply]
I see the same thing - looks like the upstream firmware repo changed branch names and likely the open PR is all that's needed. Not an iOS dev myself so not sure there's a way to change the app code without pushing an update anyways (i.e. having a maintainer approve a pr). 

If the maintainers aren't sure about the recent PR perhaps they may be ok with adding a feature to target a specific firmware url? Maybe a bug report will get eyes on it if the discussions aren't monitored much

**Reply by devinhenkel** [reply]
Cool. Thanks.

**Reply by MagicTrevor** [reply]
Unfortunately there is no current iOS maintainer.

**Reply by aafishman** [reply]
That's unfortunate. I see a few commits in the last month or two from @mig5 @ocdtrekkie and @poormusic2001 - the current PR might be checkable by anyone who has done any work on any of the other os projects since it just affects firmware. Not sure about how merging ultimately affects the App Store app tho

**Reply by ocdtrekkie** [reply]
The primary issue is that publishing to the app store requires someone with an Apple computer.

**Reply by devinhenkel** [reply]
Do you need to be part of the same org? I have a Mac, but haven't done AppStore pushes before.

**Reply by xan-m** [reply]
Yes, you need to be a member of Pine64’s Apple developer account (which I am) with current developer signing certificates (which I’m working with Pine64 to get).

**Reply by devinhenkel** [reply]
Okay. Let me know if I can help. Otherwise, thanks for tackling this.

**Comment by xan-m** [comment]
Apologies for the delay -- I'm working full time and pursuing an MS degree so my dance card is pretty full these days! 

I saw the pull request to update the branch name and merged it. I'll try to get a build sent up to the App Store today.

Thank you all for your patience!

**Reply by aafishman** [reply]
Wow appreciate it! I didn't tag you since I figured transferring the project was your way of saying 'I'm too busy' haha (seems that's mostly right 😅). 

I don't have a Mac on-hand these days for builds (and haven't ever pushed to app store) but I'll see what it takes to set up cd in GitHub to the App Store soon so you (or anyone) don't have to deal with that step. At a glance it's not too bad, I see a few templates that mostly need a config file or two, the developer cert, etc. looks like for public repos we can use GitHub hosted (free) runners too

**Comment by xan-m** [comment]
Hey all — this should be fixed in v1.0.2, which hit the App Store about 12 hours ago. Please update your app and let me know if you’re still seeing the problem.

**Comment by devinhenkel** [comment]
Worked for me. Reloaded 1.11 and it went smoothly. Thanks!D. Fat-fingered on an iOS device. On Nov 5, 2022, at 9:10 AM, xan-m ***@***.***> wrote:﻿
Hey all — this should be fixed in v1.0.2, which hit the App Store about 12 hours ago. Please update your app and let me know if you’re still seeing the problem.
